control_plane/src/pageserver.rs

```diff
@@ -419,6 +419,11 @@ impl PageServerNode {
                     .map(|x| x.parse::<bool>())
                     .transpose()
                     .context("Failed to parse 'trace_read_requests' as bool")?,
+                eviction_policy: settings
+                    .get("eviction_policy")
+                    .map(|x| serde_json::from_str(x))
+                    .transpose()
+                    .context("Failed to parse 'eviction_policy' json")?,
```

Comment:
Ok to require user to pass JSON here?

Reply:
So this is json within json, why? Aah, unsure if I've ever used this neon_local functionlity. Does seem weird but what are the alternatives?

Reply:
Understood the point raised here a bit late and started wondering if neon_local should at *all* parse the payload, except to check that it's JSON.

However using from_str here might not be great; better would be `Box<serde_json::value::RawValue>` then this property would not have to be a string containing json in a json document but just a arbitrary json subtree string. If we do have full control of the object, this could also be `#[serde(borrowed)] name: &'a RawValue`.

Reply:
Agree in principle on the `RawValue` proposal, but, neon_local uses the `pageserver_api` models directly here, and I think having the proper type in the API model is useful.

So, not doing that in this PR

Reply:
Aah ok this is a darker corner I haven't yet ran into :) Yes agreed.

> Well, I don't want to spend time on implementing a CLI-friendly representation of the eviction policy variants here. IMO there should be a serde_cli crate that standardizes such stuff.

yeah this is a weird ... middle ground. Didn't check, but I'm pretty sure our argv parser *could* be hooked to parse additional parts as serde deserialized values by turning them into `HashMap<&str, &str>` which we already have but perhaps ... yeah. Well, we can change this in future.

This could use a comment for the above reasoning just as well, so the picking up would be easier.